boards/risc-v/esp32c3/esp32c3-devkit/scripts/Make.defs

```diff
@@ -42,7 +42,13 @@ ifneq ($(CONFIG_DEBUG_NOOPT),y)
 endif
 
 ARCHCFLAGS = -fno-common -ffunction-sections -fdata-sections -msmall-data-limit=0
-ARCHCXXFLAGS = $(ARCHCFLAGS) -fno-exceptions -fcheck-new -fno-rtti
+ARCHCXXFLAGS = $(ARCHCFLAGS) -std=c++17
+ifneq ($(CONFIG_CXX_EXCEPTION),y)
+  ARCHCXXFLAGS += -fno-exceptions -fcheck-new
+endif
+ifneq ($(CONFIG_CXX_RTTI),y)
+  ARCHCXXFLAGS += -fno-rtti
+endif
```

Comment:
@gustavonihei this is common option, how about we move them to Toolschain.defs to avoid the duplication?

Reply:
Done, `CXXFLAGS` is now initialized on `Toolchain.defs` file of each architecture.